### DISCUSSION TITLE
Items will operate if contained item has required item in subcontained item, without degrading subcontained item's contained item.

### DISCUSSION THREAD

**Discussion by ShadowDragon8685** [FakeFishGames/Barotrauma#12249] [Code questions and modding support]
That sounds like a mouthful, let me explain:

Assume you have a tool that required a battery to run, and use a Subcontained item to attach a flashlight to it. (EK mods' arc welder, for example.)

The arc welder runs on batteries. The flashlight _also_ runs on batteries. If you attach a flashlight to an arc welder, _do not put a battery in the arc welder,_ and use the arc welder, the arc welder _works,_ and it _does not_ consume the flashlight's battery.

**Comment by BiffW** [comment]
Add a slot limitation to `RequiredItem` defined in arc welder to only use the battery in its own container.
For example: `<RequiredItem items="mobilebattery" type="Contained" TargetSlot="0" />`
